pkg/kubelet/kubelet.go

```diff
@@ -1597,7 +1597,14 @@ func (kl *Kubelet) syncPod(o syncPodOptions) error {
 				if err := kl.podManager.DeleteMirrorPod(podFullName); err != nil {
 					glog.Errorf("Failed deleting mirror pod %q: %v", format.Pod(mirrorPod), err)
 				} else {
-					deleted = true
+					opts := metav1.GetOptions{}
+					_, err := kl.kubeClient.Core().Nodes().Get(string(kl.nodeName), opts)
```

Comment:
I feel like this check should go next to the `CreateMirrorPod` call, instead of here?

Also, is there not a cached version of this? Doesn't the Kubelet know whether it is registered as a node with the API server?

Reply:
> I feel like this check should go next to the `CreateMirrorPod` call, instead of here?

It should be here. Because we are going to delete static pods for deleted nodes, not create them.

> Also, is there not a cached version of this? Doesn't the Kubelet know whether it is registered as a node with the API server?

We cannot identify from the cached version whether the node has been deleted from API server.

Reply:
> It should be here. Because we are going to delete static pods for deleted nodes, not create them.

I don't understand this comment. The pod has been deleted, it seems weird to lie that it hasn't. Furthermore, if syncpod is called again it will be recreated since this block will be skipped.

> We cannot identify from the cached version whether the node has been deleted from API server.

Use `kl.GetNode()`. Also do we need to check the `DeletionTimestamp`?

/cc @yujuhong

Reply:
@tallclair Make sense. Will update it. Thanks.